docs/sha2-256-chunked.md

```diff
@@ -0,0 +1,42 @@
+# sha2-256-chunked
+
+## DRAFT multihash codec 0xb510
+
+> Hash of concatenated SHA2-256 digests of 8*2^n MiB source chunks; n = source_size / (10000 * 8MiB)
+
+This variant of sha2-256 is designed to enable very large files
+to be efficiently uploaded and hashed in parallel using fixed size chunks
+(8MiB to start with), with the final result being a "top hash"
+that doesn't depend on the upload order.
+
+The algorithm has an upper limit of 10,000 hashes.
+If the file is larger than 80,000 MiB,
+it will double the block size until the number of blocks
+is under that limit.
+
+```pseudocode
+n = floor(source_size / (10_000 * 8 MiB))
+chunk_size = 8*2^n MiB
+```
```

Comment:
@drernie: this is actually wrong - we need `log` to calculate `n`, since we're doing `2^n` later. Though in the Python code, we're just using a loop cause it's simpler (and that's what boto3 does).

Simple example of the failure:

Let's say the file size would result in 32 * 10,000 parts using 8MiB chunks. The correct solution is to double the chunk size 5 times, resulting in 32*8MiB chunks. However, what we end up is:

```python
>>> MiB = 2**20
>>> source_size = 10_000 * 8 * MiB * 32
>>> n = math.floor(source_size / (10_000 * 8 * MiB))
>>> chunk_size = 8 * 2**n * MiB
>>> 
>>> source_size
2684354560000
>>> chunk_size
36028797018963968
```

Chunk size is bigger than the source size...

Reply:
Doh! Good catch.  Is this better?
```python
> Hash of concatenated SHA2-256 digests of 8*2^n MiB source chunks
> where n = log(source_size / (10^4 * 8MiB))

n = floor(log(source_size / (10_000 * 8 MiB)))
```

Reply:
```suggestion
n = ceil(log2(source_size / (10_000 * 8 MiB)))
```